core/ir/riscv64/encode.c

```diff
@@ -37,7 +37,7 @@
 /* clang-format off */
 const char *const reg_names[] = {
     "<NULL>", "<invalid>",
-    "zero", "ra", "sp", "gp",  "tp",  "t0", "t1", "t2", "s0", "fp", "s1", "a0",
```

Comment:
Why did we need to remove the s0 entry? It creates a mis-match with the DR_REG_ enum:
https://github.com/DynamoRIO/dynamorio/blob/4ef5dfef61b4e8b177f353427acd57974d51240e/core/ir/opnd_api.h#L1171

Reply:
Because both s0 and for are aliases for the X8 register. For the string name i had to choose one, otherwise registers above x8 would have a wrong name printed.

Reply:
Ah okay. Thanks.